src/python/WMCore/Storage/DeleteMgr.py

```diff
@@ -210,32 +209,40 @@ def __call__(self, fileToDelete):
             msg = "Unable to delete file:\n"
             msg += fileToDelete['LFN']
             raise StageOutFailure(msg, **fileToDelete)
-
-    def deleteLFN(self, lfn, override=None):
+    
+    def deleteLFN(self, lfn, stageOut_rfc=None):
         """
         deleteLFN
-
-        Given the lfn and local stage out params, invoke the delete
-
-        if override is used the follwoing params should be defined:
-        command - the stage out impl plugin name to be used
-        option - the option values to be passed to that command (None is allowed)
-        lfn-prefix - the LFN prefix to generate the PFN
-        phedex-node - the Name of the PNN to which the file is being xferred
+        Given the lfn and an stageOut Rucio file catalog pair or override config, invoke the delete
+        lfn: logical file name
+        stageOut_rfc: a pair fo stageOut and correponding Rucio file catalog, required when no override provided 
+        self.overrideConf: the follwoing params should be defined for override
+            command - the stage out impl plugin name to be used
+            option - the option values to be passed to that command (None is allowed)
+            lfn-prefix - the LFN prefix to generate the PFN
+            phedex-node - the Name of the PNN to which the file is being xferred
         """
+        if not self.overrideConf:
+            from WMCore.Storage.StageOutMgr import searchRFC 
```

Comment:
I can not move this to top because of circular imports

Reply:
Just a thought. How about moving this function (searchRFC) under the RucioFileCatalog then?

Reply:
I would rather to keep it as is since this is just the wrapper for matchLFN of RucioFileCatalog (adding some safeguard checks on top of matchLFN). It is transferred from master of this one:
https://github.com/nhduongvn/WMCore/blob/87ea437f508308b5e1d544a234a80e2a9911d5c1/src/python/WMCore/Storage/StageOutMgr.py#L374
I add it to the list of unchanged methods in the PR description

Reply:
And the previous implementation (master) apparently has a duplication of that function in 2 different modules, which isn't great as well.

Could you please add a comment like `# FIXME` saying that there is a circular import and that's why the import is placed in the middle of the module? At some point we have to refactor and properly separate it.

Reply:
Yes, I added this to my local copy here. It is too late to update github now since I already pushed and made a pull request. Will be included in the next push.

Reply:
Following up in the newer PR:

https://github.com/dmwm/WMCore/pull/11869/files#diff-771fca6840b96ebe5d47dd68b4d32ec9e75263a8cd641ebff965e2bac1d962f4